src/TestProblem/ELBDM/PlaneWave/Init_TestProb_ELBDM_PlaneWave.cpp

```diff
@@ -0,0 +1,294 @@
+#include "GAMER.h"
+#include "TestProb.h"
+
+
+
+// problem-specific global variables
+// =======================================================================================
+static double PWave_Lambda;        // plane wave wavelength
+static double PWave_Amp;           // plane wave amplitude
+static double PWave_Phase0;        // plane wave phase constant
+static int    PWave_XYZ;           // plane wave direction (0/1/2/3 --> x/y/z/diagonal)
+static int    PWave_LSR;           // plane wave direction (<0/0/>0 --> Left-moving/Standing/Right-moving)
+
+static double PWave_Period;        // plane wave period
+static double PWave_WaveK;         // plane wave wavenumber
+static double PWave_WaveW;         // plane wave angular frequency
+static double PWave_PhaseV;        // plane wave phase velocity
+static double PWave_GroupV;        // plane wave group velocity
+// =======================================================================================
+
+static void OutputError();
+
+
+
+
+//-------------------------------------------------------------------------------------------------------
+// Function    :  Validate
+// Description :  Validate the compilation flags and runtime parameters for this test problem
+//
+// Note        :  None
+//
+// Parameter   :  None
+//
+// Return      :  None
+//-------------------------------------------------------------------------------------------------------
+void Validate()
+{
+
+   if ( MPI_Rank == 0 )    Aux_Message( stdout, "   Validating test problem %d ...\n", TESTPROB_ID );
+
+
+// errors
+#  if ( MODEL != ELBDM )
+   Aux_Error( ERROR_INFO, "MODEL != ELBDM !!\n" );
+#  endif
+
+#  ifdef GRAVITY
+   Aux_Error( ERROR_INFO, "GRAVITY must be disabled !!\n" );
+#  endif
+
+#  ifdef PARTICLE
+   Aux_Error( ERROR_INFO, "PARTICLE must be disabled !!\n" );
+#  endif
+
+#  ifndef FLOAT8
+   Aux_Error( ERROR_INFO, "FLOAT8 must be enabled !!\n" );
+#  endif
+
```

Comment:
If you decide to use periodic BC, add a sanity check here.

Reply:
Done.